src/viam/components/resource_manager.py

```diff
@@ -39,6 +42,12 @@ def register(self, component: ComponentBase):
             if hasattr(subtype, "get_resource_name"):
                 rn = subtype.get_resource_name(component.name)  # type: ignore
                 rnames[rn] = component
+            if rn.name.__contains__(":"):
+                short_name = rn.name.split(":")[-1]
+                if short_name in self._short_to_long_name and rn not in self._short_to_long_name[short_name]:
```

Comment:
On line 17 you specify the type of `_short_to_long_name` as `Dict[str, ResourceName]` but here, you're using it as `Dict[str, List[ResourceName]]`. Please update the typedef

Reply:
Thanks for catching that!